chia/types/name_puzzle_condition.py

```diff
@@ -18,5 +18,6 @@ class NPC(Streamable):
     def condition_dict(self):
         d: Dict[ConditionOpcode, List[ConditionWithArgs]] = {}
         for opcode, l in self.conditions:
+            assert opcode not in d
```

Comment:
my understanding is that we don't allow having duplicates here, because the second entry would clobber the first in the dictionary

Reply:
Good looking out for adding the assert, but ...

If I understand correctly, it could be legitimate to have e.g. multiple `ASSERT_COIN_ANNOUNCEMENT`

Reply:
right, this is a list of lists, the top level list is really a dictionary and so shouldn't allow duplicate keys. My comment here is about how this constrain isn't really expressed in the data structure, since it's a list of lists.

I have a patch in the works which simplifies the data structure of the conditions, making it more concrete. This patch is one, somewhat, independent part of that larger patch.